static/fetch.js

```diff
@@ -1,31 +1,179 @@
 const RELOAD_TIME = 3000; 
-const PERSISTENCE_TIME = 180000;
+const PERSISTENCE_TIME_COMPLETE = 180000;
+const PERSISTENCE_TIME_PENDING = 300000;
+const PERSISTENCE_TIME_ERROR = 300000;
+
+const PrintJob = {
+    PARSE(code) {
+        for (let key in PrintJob) {
+            if (PrintJob[key].CODE === code) return PrintJob[key]
+        }
+    },
+    COMPLETED: {
+        CODE: 0,
+        ATTR: 'completed',
+        MSG: 'SUCCESS'
+    },
+    PENDING: {
+        CODE: 1,
+        ATTR: 'pending',
+        MSG: 'PENDING'
+    },
+    FILE_ERROR: {
+        CODE: 2,
+        ATTR: 'error',
+        MSG: 'FAILURE'
+    },
+    QUOTA_LIMIT: {
+        CODE: 3,
+        ATTR: 'error',
+        MSG: 'FAILURE'
+    },
+    JOB_ERROR: {
+        CODE: 4,
+        ATTR: 'error',
+        MSG: 'FAILURE'
+    }
+}
+
+
+function _element(tag, classList, attributes, innerText){
+    let temp = document.createElement(tag);
+    if (classList && !(classList instanceof Array) && typeof classList === 'string') classList = [classList];
+    else classList = [];
+    if (attributes && attributes instanceof Array) {
+        let temp = {};
+        attributes.forEach(item => {
+            temp[item] = true;
+        });
+        attributes = temp;
+    } else if (attributes && typeof attributes === 'string') {
+        attributes = {[attributes]: true}
+    } else {
+        attributes = {};
+    }
+    classList.forEach(className => temp.classList.add(className));
+    for (let key in attributes) {
+        temp.setAttribute(key, attributes[key] && typeof attributes[key] === 'boolean' ? '' : String(attributes[key]));
+    }
+    if (innerText) temp.innerText = innerText;
+    return temp;
+}
 
 class Job {
-    constructor(parent, printer_name, username, date) {
-        this.date = new Date(date*1000);
-        this.parent = parent;
-        this.printer_name = printer_name;
-        this.username = username;
-        this.element = this.createElement();
-        this.symbol = Symbol();
+    constructor(printer, job) {
+        this.printer = printer;
+        this.username = job.username;
+        this.id = job.id;
+        this.last_updated = new Date(job.last_updated*1000);
+        this.createElement();
+        this.update(job);
+    }
+    update(job) {
+        this.status = job.status;
+        this.last_updated = new Date(job.last_updated*1000);
+        let stat = PrintJob.PARSE(this.status);
+        this.entry.status.innerText = stat.MSG;
+        this.entry.status.setAttribute('status', stat.ATTR);
+    }
+    getTime() {
```

Comment:
Please don't do this. JavaScript [already knows how to do this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString).